pkg/infra/teams/team3/agent3.go

```diff
@@ -29,7 +29,7 @@ func (a *AgentThree) UpdateInternalState(baseAgent agent.BaseAgent, _ *commons.I
 	AS := baseAgent.AgentState()
 	view := baseAgent.View()
 	// Initialise utils
-	if view.CurrentLevel() == 1 {
+	if view.CurrentLevel() == 0 {
```

Comment:
This seems to be what caused the issue - you were simultaneously initialising with `InitUtility` and performing reassignment in `UpdateTotalUtility`

Reply:
Aha, I see. But how wasn't there a problem before? Cause that's been here for a while...

Reply:
No, this is true, you're right. The issue seems to be the concurrent write and read of this data. I'm not sure what was added since, as all I've done is prevent the initialisation. It would be interesting to discuss what changes were made